File: Layer.js
class Layer {
  constructor(neurons, activation, inputLength){
    this.neurons = neurons
    this.operations = []
    this.inputLength = inputLength
    this.activation = activation
  }
  setup(){
    for(var i = 0;i < this.neurons;i++){
      var connections = []
      for(var j = 0;j < this.inputLength;j++){
          connections.push(new Connection())
      }
      this.operations.push({
        neuron: new Neuron(),
        connections: connections
      })
    }
  }
  forward(input){
    var output = []
    
    for(var i = 0;i < input.length;i++){
      for(var j = 0;j < this.operations.length;j++){
        var c = []
        for(var k = 0;k < this.operations[j].connections.length;k++){
          c[k] = this.operations[j].connections[k].forward(input[i])
        }
        output[j] = this.operations[j].neuron.forward(c)
      }
    }
    
    return output
  }
}

File: Layer.py
from Neuron import Neuron
from Connection import Connection

class Layer:
  def __init__(self, numberOfNeurons, activation, inputLength):
    self.numberOfNeurons = numberOfNeurons
    self.operations = []
    self.inputLength = inputLength
    self.activation = activation
  
  def setup(self):
    for i in range(self.numberOfNeurons):
        connections = []
        for j in range(self.inputLength):
            connections.append(Connection())
        
        self.operations.append({
            'neuron': Neuron(),
            'connections': connections
        })
    
  
  def forward(self, input):
    output = []
    
    for inputItem in input:
        for operation in self.operations:
            c = []
            for connection in operation['connections']:
                c.append(connection.forward(inputItem))
            
            output.append(operation['neuron'].forward(c))
        
    
    return output

